packages/build/src/plugins/plugin_conditions.ts

```diff
@@ -81,7 +83,10 @@ const siteDependencyTest = async function ({
 
   try {
     // if this is a range we need to get the exact version
-    const packageJsonPath = await resolvePath(`${dependencyName}/package.json`, buildDir)
+    const packageJsonPath = await resolvePath(
+      `${dependencyName}/package.json`,
+      packagePath ? join(buildDir, packagePath) : buildDir,
```

Comment:
nit - we seem to mostly be using `join(buildDir, packagePath || '')` or `packagePath ?? ''`